### DISCUSSION TITLE
Shared Config Across Multiple Projects

### DISCUSSION THREAD

**Discussion by Shackelford-Arden** [ansible/ansible-lint#4162] [Q&A] [answered]
## Scenario

My team and I manage a set of 20-30 Ansible roles. We'd like to have them all share the same ansible-lint config. While we can relatively easily manage the config file in each repo w/ some additional tooling, we'd like to find a better way, if one exists.

## Question

Is there a built-in method w/ `ansible-lint` to have it reference a git repo or URL instead of having to have a local file present in each repo?

## Existing Method

What I'm doing now is essentially setting up each repo to have CI download the shared config and point `ansible-lint` to it.

#### Local Reproduction

When team members are developing roles locally, they have to manually manage pulling down the shared config each time they need it, or setting themselves up w/ some scripts/etc to run `ansible-lint` locally on the role they're developing.

Would be really handy to be able to have it where we can tell `ansible-lint` to look at a repo/URL.

**Comment by sean-m-sullivan** [comment, marked as answer]
There is no one answer, the best answer would be to put it in a CI template that is shared and injected into the project when the linting runs. Though this prevents it being run locally.

**Reply by Shackelford-Arden** [reply]
Yeah, that's kind of what I figured, which is why I'm now working on doing just as you've described.

Kind of makes me think that opening a feature request might be the only thing I could really do at this point.

**Reply by cavcrosby** [reply]
Yea, I think it would be worth raising an issue/feature request to see if the maintainers believe this would be worth integrating.
